core/dbt/parser/schema_yaml_readers.py

```diff
@@ -669,16 +669,25 @@ def _generate_saved_query_config(
 
         return config
 
-    def _get_export_config(self, unparsed: UnparsedExportConfig) -> ExportConfig:
-        return ExportConfig(
-            export_as=unparsed.export_as,
-            schema_name=unparsed.schema,
-            alias=unparsed.alias,
+    def _get_export_config(
+        self, unparsed: Dict[str, Any], saved_query_config: SavedQueryConfig
+    ) -> ExportConfig:
+        # Combine the two dictionaries using dictionary unpacking
+        # the second dictionary is the one whose keys take priority
+        combined = {**saved_query_config.__dict__, **unparsed}
+        # `schema` is the user facing attribute, but for DSI protocol purposes we track it as `schema_name`
+        if combined.get("schema") is not combined.get("schema_name") is None:
```

Comment:
I'm finding this predicate a bit tricky to read: `A is not B is None`? is something missing after `is not`?

Reply:
That is indeed a bit wonky, I think something got shortened 😬 Will revisit, and maybe write an additional test

Reply:
Feels like I when I write a sentence, and then edit it multiple times, and then it becomes incoherent 🙃

Reply:
Fixed in [dd3be16](https://github.com/dbt-labs/dbt-core/pull/9065/commits/dd3be1640c3ecd65598ec8caa356c5580df61221)